### ISSUE TITLE
Fork attacks on the Filecoin blockchain

### ISSUE THREAD

**Issue by ChainDotInfo** [filecoin-project/lotus#1772]
**Describe the bug**
On the Testnet phase 2, I found my (t01043) mined blocks are mostly invalidated by forks. As seen below, it happens 5 times an hour. 

![image](https://user-images.githubusercontent.com/55821391/82139738-afad3a00-985c-11ea-99ed-1eed7638c861.png)

I think it's abnormal. I doubt large miners are doing 51% attacks on the Testnet, to create a blockchain where only their blocks exist, avoiding sharing rewards with other miners.

I think it’s a drawback to the consensus. Large miners can deliberately keep creating blocks on an alternative tipset, in which blocks from other miners are excluded. In this way, he can not only exclude other blocks in the previous tipset, but also the current tipset. And he will not be punished by doing this.

It’s a kind of 51% attack based on storage power. The difference is, using storage power does not have any additional cost.

In PoS, such attacks can be easily avoided by preventing miners from generating two blocks on the same height. But in Filecoin, that is not enough. Because miners are easy to change the info of the previous height in Filecoin, which is not realistic in PoS.

In conclusion, I think it’s easy to do fork attacks on Filecoin. You can simply try different tipsets by removing some of the blocks from the tipset, and try to generate a new block on these different tipsets. If you are able to generate a relatively heavy-weighted block, then broadcast it. It only needs to cost additional storage power, which is actually some reads of the disk.

**Comment by chengwds** [comment]
There are many miners, only a few can work normally. Is this normal
